smoke/LeafDevice/details/Details.cs

```diff
@@ -30,14 +36,33 @@ public class Details
             string eventhubCompatibleEndpointWithEntityPath,
             string deviceId,
             string certificateFileName,
-            string edgeHostName
+            string edgeHostName,
+            Option<UpstreamProtocolType> upstreamProtocol
         )
         {
             this.iothubConnectionString = iothubConnectionString;
             this.eventhubCompatibleEndpointWithEntityPath = eventhubCompatibleEndpointWithEntityPath;
             this.deviceId = deviceId;
             this.certificateFileName = certificateFileName;
             this.edgeHostName = edgeHostName;
+
+            switch (upstreamProtocol.GetOrElse(UpstreamProtocolType.Amqp))
+            {
+                case UpstreamProtocolType.Amqp:
+                case UpstreamProtocolType.Mqtt:
+                    this.serviceClientTransportType = ServiceClientTransportType.Amqp;
+                    this.eventHubClientTransportType = EventHubClientTransportType.Amqp;
+                    this.deviceClientTransportType = DeviceClientTransportType.Mqtt;
```

Comment:
It probably doesn't make sense to call it `--upstream-protocol` for `LeafDevice`. It made sense for `Quickstart` because the value was also used as the `UpstreamProtocol` env var (in addition to choosing the transport type for the various C# clients).

Here it's only used to select between websockets and not-websockets for the C# clients. It'd be better to just have a `--use-websockets` bool.

Reply:
Ah, ok, it was just hard-coded before to always create a device client with MQTT. I guess all you're really trying to do is allow the user to specify "with WebSocket" or "without WebSocket", so adding AMQP & MQTT variants confused me--you aren't really allowing people to choose AMQP or MQTT.

Your code works fine, but I'd consider either changing the enum (and the cmd line arg) to WithWebSocket/WithoutWebSocket, or make the device client protocol exactly match the option the caller passed in.